src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
```

Comment:
I'm not fond of sending an object here, we would add usesless treatements that will slow down our Algorithms, we should stuck to arrays as basic inputs, what do you think?
Plus, i don't like having csvDataset in this project either, the goal of this project is to provide machine learning alogrithms, if someone want to aggregate data from a CVS, an XLS, a JSON ... there are plenty of libraries to do that, what do you think of removing these classes and focus solely on ML Algorithms?